internal/state/state.go

```diff
@@ -61,13 +61,7 @@ func getDefaultAccount(accountList []accounts.Account) (*accounts.Account, error
 	if len(accountList) == 0 {
 		return nil, errNoAccounts
 	} else if len(accountList) > 1 {
-		// If an account was provided via environment variables, use it.
-		for _, acct := range accountList {
-			if acct.Source == accounts.AccountSourceEnvironment {
-				return &acct, nil
```

Comment:
A bit of digging into this, it looks like the only place we make accounts is in the Credentials providers which loads the file or keychain credentials. Correct my if I'm wrong, but that should mean we never see `AccountSourceEnvironment` account types here.

Reply:
Totally correct, that's why it was ok to remove this code.